### PR TITLE
update the data prep repo to accept a directory of input jsonls

### PR DESCRIPTION
**Pull request by snova-zoltanc**
## Summary
Update input_file_path argument to input_path, allow it to take a directory of files as an input. 
Complete issue: https://github.com/sambanova/generative_data_prep/issues/114 

## PR Checklist
- [X] My PR is less than 500 lines of code
- [X] I have added sufficient comment as docstrings in my code
- [X] I have made corresponding changes to the documentation
- [ ] I have written unit-tests to test all of my code

### CODE REVIEW

File: generative_data_prep/__main__.py [hunk 1]
```diff
@@ -302,12 +304,40 @@ def get_categories(categories_path: str):
...
+    """Processes a directory containing JSONL files and combines them into a single output file.
+
+    Args:
+        input_path (str): The path to the directory containing the input JSONL files.
+
+    Returns:
+        str: The path to the combined output JSONL file. If there is only one JSONL file
+             in the directory, returns the path to that file directly without combining.
+    """
+    input_jsonl_files = glob(f"{input_path}/*.jsonl")
```
Comment (vmly) [inline generative_data_prep/__main__.py:317]: Do we need to raise an error if no files are found here ?

Comment (vmly) [inline generative_data_prep/__main__.py:317]: There is a risk of path//*.jsonl if the input path provided ends with a /

I would prefer using os.path.join or pathlib.Path.glob() here

**Review by vmly — COMMENTED** [review]
Thanks for the quick update 😃 . I have added a few minor comments. Otherwise, Looks good to me. 

One more question, Is there a history or release notes where we need document this breaking change to the data prep script ?

File: generative_data_prep/utils/utils.py [hunk 2]
```diff
@@ -57,7 +57,7 @@ def data_prep_arg_builder(parser: argparse.ArgumentParser):
     Args:
         parser (argparse.ArgumentParser): parser to add arguments to
     """
-    parser.add_argument("--input_file_path", type=str, required=True, help="The input jsonl file path.")
+    parser.add_argument("--input_path", type=str, required=True, help="The input jsonl file path.")
```
[the remark this answers is not in the document]
Comment (snova-zoltanc) [reply generative_data_prep/utils/utils.py:60]: Fixed

File: generative_data_prep/__main__.py (hunk 1 above)
Comment (snova-zoltanc) [reply generative_data_prep/__main__.py:317]: Done

Comment (snova-zoltanc) [reply generative_data_prep/__main__.py:317]: Fixed using pathlib

**Comment by snova-zoltanc** [comment]
>  history or release notes where we need document this breaking change
Not sure what this means, We don't have any specific release notes. Maybe we could add a specific failiure that tells the user that the flag name has changed?

**Review by snova-edwardm — DISMISSED** [review]

File: tests/examples/directory_input/pipelined_directory_input/metadata.yaml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+max_batch_size_dev: null
+max_batch_size_train: 5
+max_seq_length: 1024
+number_of_dev_files: 0
+number_of_test_files: 0
+number_of_training_files: 32
+token_type_ids: true
+tokenizer_model_type: <class 'transformers.models.gpt2.configuration_gpt2.GPT2Config'>
```
Comment (snova-jonathanl) [inline tests/examples/directory_input/pipelined_directory_input/metadata.yaml:8]: should this be a simpler string for the class name?

**Review by snova-jonathanl — APPROVED** [review]
Generally lgtm!

File: tests/examples/directory_input/pipelined_directory_input/metadata.yaml (hunk 3 above)
Comment (snova-zoltanc) [reply tests/examples/directory_input/pipelined_directory_input/metadata.yaml:8]: Yeah good point, this is not relevant to PR though so I will keep it as a note when I work on metadata 🙇

**Review by snova-jonathanl — APPROVED** [review]
re-badging, lgtm!

**Pull request merged**